src/gwt/src/org/rstudio/core/client/StringUtil.java

```diff
@@ -944,6 +944,40 @@ public static String getExtension(String string)
       return getExtension(string, 1);
    }
    
+   public static String getCssIdentifier(String string)
+   {
+      // Each character must be one of the following:
+      // alphanumeric, an ISO 10646 character U+00A0 or higher, a hyphen, or an underscore.
+      // Identifiers cannot start with a hyphen, two hyphens, or a hyphen followed by a digit.
+      // This implementation considers escaped characters invalid.
+      // If an invalid character is found, it is replaced with an 'x'.
+      
+
+      // return the string if it's already valid,
+      // otherwise replace invalid characters with 'x'
+      Pattern pattern =
+         Pattern.create("(^[a-zA-Z][a-zA-Z0-9\\-\\_]+$)|(^-[a-zA-Z][a-zA-Z0-9\\-\\_]+$)");
```

Comment:
I think this regex would be more simply expressed with a leading `-?` (then it won't be necessary to duplicate the whole alphanumeric test)